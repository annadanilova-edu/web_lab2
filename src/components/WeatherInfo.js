import React from 'react';

export default class WeatherInfo extends React.Component {
    render() {
        return (
            <table>
                <tbody>
                <tr>
                    <td>Ветер</td>
                    <td style={{'textAlign': "end"}}>{this.props.windDirection}, {this.props.windSpeed} m/s</td>
                </tr>
                <tr>
                    <td>Облачность</td>
                    <td style={{'textAlign': "end"}}>{this.props.clouds}</td>
                </tr>
                <tr>
                    <td>Давление</td>
                    <td style={{'textAlign': "end"}}>{this.props.pressure} hpa</td>
                </tr>
                <tr>
                    <td>Влажность</td>
                    <td style={{'textAlign': "end"}}>{this.props.humidity} %</td>
                </tr>
                <tr>
                    <td>Координаты</td>
                    <td style={{'textAlign': "end"}}>[{this.props.lat}, {this.props.lon}]</td>
                </tr>
                </tbody>
            </table>
        )
    }
}