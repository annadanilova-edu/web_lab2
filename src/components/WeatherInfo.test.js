import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from "redux-mock-store"
import {configure, shallow} from "enzyme";
import WeatherInfo from './WeatherInfo';

configure({ adapter: new Adapter() });

describe('WeatherInfo', () => {
    it('renders correctly', () => {
        const mockStore = configureMockStore();
        const store = mockStore({
            windSpeed: 0.47,
            windDirection: "north",
            clouds: "broken clouds",
            pressure: 1016,
            humidity: 93,
            lat: 35,
            lon: 139
        });
        const wrapper = shallow(<WeatherInfo store={store}/>);
        expect(wrapper.firstChild).toMatchSnapshot();
    })
});