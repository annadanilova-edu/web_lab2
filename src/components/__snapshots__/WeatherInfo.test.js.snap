// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo renders correctly 1`] = `undefined`;
